relayer/pkg/chainlink/txm/txm.go

```diff
@@ -46,21 +50,29 @@ type starktxm struct {
 	queue   chan Tx
 	ks      keys.Keystore
 	cfg     Config
+	nonce   keys.NonceManager
 
-	// TODO: use lazy loaded client
-	client    *starknet.Client
-	getClient func() (*starknet.Client, error)
+	client  *utils.LazyLoad[*starknet.Client]
+	txStore ChainTxStore
 }
 
 func New(lggr logger.Logger, keystore keys.Keystore, cfg Config, getClient func() (*starknet.Client, error)) (StarkTXM, error) {
-	return &starktxm{
-		lggr:      logger.Named(lggr, "StarknetTxm"),
-		queue:     make(chan Tx, MaxQueueLen),
-		stop:      make(chan struct{}),
-		getClient: getClient,
-		ks:        keystore,
-		cfg:       cfg,
-	}, nil
+	txm := &starktxm{
+		lggr:    logger.Named(lggr, "StarknetTxm"),
+		queue:   make(chan Tx, MaxQueueLen),
+		stop:    make(chan struct{}),
+		client:  utils.NewLazyLoad(getClient),
+		ks:      keystore,
+		cfg:     cfg,
+		txStore: ChainTxStore{},
+	}
+	client, err := txm.client.Get()
```

Comment:
good question!
`Get()` fetches a client based on the chains/nodes implementation. so if there's an error you can clear the client and retrieve another once from the list (assuming there are more than one implemented)

this prefetching is mostly just an optimization so the nonce manager can use it easily

Reply:
i think that makes sense! (just double checking my understanding - you're suggesting we could just go through all the clients to find a valid one at the beginning, and then we don't need lazy load)

based on that, I guess my question would be what happens if the endpoint goes down and the client starts to fail? lazy load would allow for it to switch to a different valid client